openshift2nulecule/cli/main.py

```diff
@@ -114,45 +117,50 @@ def run(self):
 
         # export project info from openshift
         exported_project = oc.export_project()
-
+        artifacts = {}
         # export images
-        if args.export_images != "none":
-            if args.export_images == "internal":
-                only_internal = True
-            elif args.export_images == "all":
-                only_internal = False
-
-            exported_project.pull_images(args.oc_registry_host,
-                                         oc.get_username(),
-                                         oc.get_token(),
-                                         only_internal)
-
-            # if registy-host is not set do not perform push
-            if args.registry_host:
-                exported_project.push_images(args.registry_host,
-                                             registry_user, registry_password,
-                                             only_internal)
-
-            exported_project.update_artifacts_images()
-
-        artifacts = exported_project.artifacts
-
-        # list of artifact for Nulecule file
-        nulecule_artifacts = []
-
-        os.makedirs(artifacts_dir)
-
-        filepath = os.path.join(artifacts_dir, "artifacts.json")
-        nulecule_artifacts.append("file://{}".format(os.path.relpath(
-            filepath, nulecule_dir)))
-        anymarkup.serialize_file(artifacts, filepath, format="json")
+        for provider in exported_project:
+            if args.export_images != "none":
+                if args.export_images == "internal":
+                    only_internal = True
+                elif args.export_images == "all":
+                    only_internal = False
+
+                exported_project[provider].pull_images(args.oc_registry_host,
```

Comment:
Like this we are pulling same image twice.
We don't need to pull images for every provider.